EIPS/eip-3712.md

```diff
@@ -0,0 +1,321 @@
+
+    event Approval(uint256 id,address indexed owner, address indexed spender, uint256 value);
+    
+    event TransferBatch(uint256[] ids,address indexed from,address indexed to,uint256[] value);
+    
+    event TransferBatchNoIndexed(uint256[] ids,address[] from,address[] to,uint256[] value);
+    
+    event ApproveBatch(uint256[] ids,address indexed from,address indexed to,uint256[] value);
+    
+    event ApproveBatchNoIndexed(uint256[] ids,address[] from,address to,uint256[] value);
+}
+```
+
+Each of the interfaces `totalSupply`,`balanceOf`,`transfer`,`allowance`,`transferFrom`,`approve(uint256 id,address spender, uint256 amount)`  introduces a new parameter id  in the ERC-20's corresponding interface's parameter list to specify a fungible token's type. 
+
+
+`approve(address spender,bool status)` is similar to ERC-1155's interface. I t allows an approval to be set uniformly for all types of tokens.
+
+`transferBatch` and `transferFromBatch` remove the parameter bytes memory data that the ERC-1155 standard uses and implement batch transactions.
+
+`transferBatchMul` implements transactions of multiple types of fungible tokens to multiple receiving addresses.
+
+`transferFromBatchMul` implements transactions of multiple types of fungible tokens from multiple sending addresses to multiple receiving addresses.
+
+
+## Rationale
+
+The ERC-20 token standard defines a single fungible token. The ERC-1155 token standard defines multiple types of tokens and specifies each token type by using a parameter `id`. By  using this parameter `id`, the ERC-1155 token standard greatly reduces the gas consumption in transactions of multiple types of tokens, compared with the ERC-20 token standard. Therefore our proposed standard uses a parameter `id` as well to specify a token type in order to reduce the gas consumption for  transactions of multiple types of tokens in one call operation via `transferBatch` or `transferFromBatch`. In our proposed st andard, each type of fungible token is specified by an id, therefore we add an interface `approve(uint256 id,address spender, uint256 amount)` for applications to set an approval for the id specified type of fungible token. In order to set approvals for multiple types of fungible tokens in minimum operations we introduce a function a`pprove(address spender,bool _status)`  to set approvals uniformly for all types of fungible tokens in one call operation. To handle a transaction of multiple types of fungible tokens from one sending address to multiple receiving addresses we introduce an interface `transferBatchMul` . To handle a transaction of multiple types of fungible tokens from multiple sending addresses to multiple receiving addresses we introduce an interface `transferFromBatchMul` .     
+
+## Backwards Compatibility
+
+Our proposed standard strictly adheres to the EIP rules and specifications, therefore our standard is backwork compatible with existing EIPs. 
```

Comment:
Please specify if you EIP is backwards compatible with other ERC standards and the Ethereum protocol.

Reply:
> Please specify if you EIP is backwards compatible with other ERC standards and the Ethereum protocol.

Thank you for your suggestion. I just submitted the changes.